File: Difficulty/Medium/[102] Binary Tree Level Order Traversal/solution.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, x):
#         self.val = x
#         self.left = None
#         self.right = None

class Solution:
    def levelOrder(self, root: TreeNode) -> List[List[int]]:
        if root is None:
            return []
        res, q, size = [], collections.deque(), 0
        q.append(root)
        while q:
            size = len(q)
            tmp = []
            while size > 0:
                ptr = q.popleft()
                if ptr.left is not None:
                    q.append(ptr.left)
                if ptr.right is not None:
                    q.append(ptr.right)
                tmp.append(ptr.val)
                size -= 1
                if size == 0:
                    res.append(tmp)
        return res

File: Difficulty/Medium/[102] Binary Tree Level Order Traversal/solution.cpp
/*
 * @lc app=leetcode id=102 lang=cpp
 *
 * [102] Binary Tree Level Order Traversal
 */

// @lc code=start
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode(int x) : val(x), left(NULL), right(NULL) {}
 * };
 */
class Solution {
public:
    vector<vector<int>> levelOrder(TreeNode* root) {
        if (root == nullptr) return vector<vector<int>>();
        vector<vector<int>> res;
        queue<TreeNode*> q;
        int size;
        q.push(root);
        while (!q.empty()) {
            size = q.size();
            vector<int> tmp;
            while(size--) {
                TreeNode* ptr = q.front();
                if (ptr->left != nullptr) q.push(ptr->left);
                if (ptr->right != nullptr) q.push(ptr->right);
                tmp.push_back(ptr->val);
                q.pop();
                if (size == 0)
                    res.push_back(tmp);
            }
        }
        return res;
    }
};
// @lc code=end

